### PR TITLE
feat!: remove Transifex calls for OEP-58 | FC-0012

### PR DESCRIPTION
**Pull request by hopewise**
## Breaking change!

This change breaks the Jenkins transifex integration which has been deprecated in favor of the new GitHub Transifex App integration as part of [OEP-58](https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0058-arch-translations-management.html#specification).

## Changes

 - Removes all direct use of `tx pull` and `tx push` commands from the micro-frontend in favor
of the [`atlas pull`](https://github.com/openedx/openedx-atlas/) command.
 - Remove source and language translations from the repositories, hence no `.json` files will be committed into the repos. 
 - `src/i18n/index.js` should export and empty array so the `make pull_translations` override it with the dynamic list of languages.
 - Remove all Transifex related `Makefile` targets and other files.

Test results
------------

 - [ ] Verify that `make extract_translations` works as expected.

<details><summary>make extract_translations test results</summary>

```
# I've run the following commands:
$ make requirements
$ make pull_translations
$ git diff

# Output of the commmands:

make extract_translations
npm ci
npm WARN prepare removing existing node_modules/ before installation

> @fortawesome/fontawesome-common-types@6.4.2 postinstall /Users/sadaftech/Documents/zeit-labs/frontend-component-footer/node_modules/@fortawesome/fontawesome-common-types
> node attribution.js

Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com
License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License)
Copyright 2023 Fonticons, Inc.


> @fortawesome/free-solid-svg-icons@6.4.2 postinstall /Users/sadaftech/Documents/zeit-labs/frontend-component-footer/node_modules/@fortawesome/free-solid-svg-icons
> node attribution.js

Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com
License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License)
Copyright 2023 Fonticons, Inc.


> @fortawesome/free-regular-svg-icons@6.4.2 postinstall /Users/sadaftech/Documents/zeit-labs/frontend-component-footer/node_modules/@fortawesome/free-regular-svg-icons
> node attribution.js

Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com
License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License)
Copyright 2023 Fonticons, Inc.


> @fortawesome/free-brands-svg-icons@6.4.2 postinstall /Users/sadaftech/Documents/zeit-labs/frontend-component-footer/node_modules/@fortawesome/free-brands-svg-icons
> node attribution.js

Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com
License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License)
Copyright 2023 Fonticons, Inc.


> @fortawesome/fontawesome-svg-core@6.4.2 postinstall /Users/sadaftech/Documents/zeit-labs/frontend-component-footer/node_modules/@fortawesome/fontawesome-svg-core
> node attribution.js

Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com
License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License)
Copyright 2023 Fonticons, Inc.


> sharp@0.32.6 install /Users/sadaftech/Documents/zeit-labs/frontend-component-footer/node_modules/sharp
> (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)

sharp: Using cached /Users/sadaftech/.npm/_cacache/_libvips/libvips-8.14.5-darwin-x64.tar.br
sharp: Integrity check passed for darwin-x64

> core-js@2.6.12 postinstall /Users/sadaftech/Documents/zeit-labs/frontend-component-footer/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"


> core-js-pure@3.33.2 postinstall /Users/sadaftech/Documents/zeit-labs/frontend-component-footer/node_modules/core-js-pure
> node -e "try{require('./postinstall')}catch(e){}"

added 1966 packages in 31.509s
# Pulling display strings from .jsx files into .json files...
rm -rf ./temp/babel-plugin-formatjs
npm run-script i18n_extract

> @edx/frontend-component-footer@1.0.0-semantically-released i18n_extract /Users/sadaftech/Documents/zeit-labs/frontend-component-footer
> fedx-scripts formatjs extract

Running with resolved config:
/Users/sadaftech/Documents/zeit-labs/frontend-component-footer/babel.config.js

# Gathering JSON messages into one file...
./node_modules/.bin/transifex-utils.js ./temp/babel-plugin-formatjs ./src/i18n/transifex_input.json

```

</details> 

Merge timeline
-----------------------

This should only be merged after [Translation Infrastructure update OEP-58](https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0058-arch-translations-management.html#specification) is fully implemented.

The timing announcement will be shared by @brian-smith-tcril on [#translations-project-fc-0012](https://openedx.slack.com/archives/C04R6TUJB7T) Open edX Slack channel.

Keep this pull request as a draft to prevent accidental merge.

### Pre-merge checklist

 - [ ] Wait for approval on https://github.com/openedx/wg-translations/issues/20

References
----------

This contribution is part of the [FC-0012 project](https://openedx.atlassian.net/l/cp/XGS0iCcQ) which is sparked by the [Translation Infrastructure update OEP-58](https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0058-arch-translations-management.html#specification).

Up-to-date project overview and details are available in the [Approach Memo and Technical Discovery: Translations Infrastructure Implementation](https://docs.google.com/document/d/11dFBCnbdHiCEdZp3pZeHdeH8m7Glla-XbIin7cnIOzU/edit#) document.

Join the conversation on [Open edX Slack #translations-project-fc-0012](https://openedx.slack.com/archives/C04R6TUJB7T).

Check the links above for full information about the overall project.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @hopewise!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by e0d** [comment]
@hopewise looks like there are some test failures.

**Comment by hopewise** [comment]
a new line added to the `src/i18n/index.js`, to allow test to pass.

**Review by OmarIthawi — CHANGES_REQUESTED** [review]
Please remove the `pull_translations` step altogether. Other MFEs will pull translations for the footer.

File: Makefile [hunk 1]
```diff
@@ -35,20 +32,18 @@ detect_changed_source_translations:
...
-	tx push -s
-	# Fetching hashes from Transifex...
-	./node_modules/@edx/reactifex/bash_scripts/get_hashed_strings_v3.sh
-	# Writing out comments to file...
-	$(transifex_utils) $(transifex_temp) --comments --v3-scripts-path
-	# Pushing comments to Transifex...
-	./node_modules/@edx/reactifex/bash_scripts/put_comments_v3.sh
-
-# Pulls translations from Transifex.
 pull_translations:
```
[the remark this answers is not in the document]
Comment (hopewise) [reply Makefile]: @OmarIthawi ok, I removed the `pull_translations `

File: src/i18n/index.js [hunk 2]
```diff
@@ -1,34 +1 @@
...
-  'fr-ca': frCAMessages,
-  ru: ruMessages,
-  uk: ukMessages,
-  'de-de': dedeCAMessages,
-  'it-it': ititCAMessages,
-  'pt-pt': ptptCAMessages,
-};
-
-export default messages;
+export default [];
```
Comment (OmarIthawi) [inline src/i18n/index.js:1]: One more fix.

```suggestion
export default {};
```

**Review by OmarIthawi — APPROVED** [review]
Looks good. I've merged one fix.

**Comment by OmarIthawi** [comment]
@timmc-edx this one is ready but pending 2U approval.

cc: @brian-smith-tcril

**Review by brian-smith-tcril — APPROVED** [review]

**Comment by OmarIthawi** [comment]
@brian-smith-tcril rebased.

**Comment by OmarIthawi** [comment]
@brian-smith-tcril rebased :)

**Comment by openedx-webhooks** [comment]
@hopewise 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-component-footer&entry.867055334=https://github.com/openedx/frontend-component-footer/pull/355&entry.1484655318=https://github.com/hopewise&entry.752974735=2023-10-21+09:02&entry.1917517419=2024-03-25+15:23&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**